node/pseudonode.py

```diff
@@ -0,0 +1,66 @@
+import time
+import uuid
+import qrcode
+import json
+import os
+import logging
+import requests
+from btclib import bip32, slip132
+
+import config
+from payments.price_feed import get_btc_value
+from payments import database
+
+if config.tor_bitcoinrpc_host is not None:
+    from gateways.tor import session
+
+
+class pseudonode:
+    def __init__(self):
+        self.api = "https://mempool.space/api"
+        self.address_counter_file = "address_counter"
+        if not database.check_address_table_exists():
+            database.create_address_table()
+
+        logging.info("Fetching blockchain info from {}".format(self.api))
+        logging.info(
+            "Next address shown to users is #{}".format(self.get_next_address_index())
+        )
+
+    def create_qr(self, uuid, address, value):
+        qr_str = "{}?amount={}&label={}".format(address.upper(), value, uuid)
+
+        img = qrcode.make(qr_str)
+        img.save("static/qr_codes/{}.png".format(uuid))
+        return
+
+    def check_payment(self, address):
+        conf_paid, unconf_paid = 0, 0
+        try:
+            r = requests.get(self.api + "/address/{}".format(address))
+            r.raise_for_status()
+            stats = r.json()
+            conf_paid = stats["chain_stats"]["funded_txo_sum"] / (10 ** 8)
+            unconf_paid = stats["mempool_stats"]["funded_txo_sum"] / (10 ** 8)
+            time.sleep(2)
+            return conf_paid, unconf_paid
+
+        except Exception as e:
+            logging.error(
+                "Failed to fetch address information from mempool: {}".format(e)
+            )
+
+        return 0, 0
+
+    def get_next_address_index(self):
+        n = database.get_next_address_index()
+        return n
+
+    def get_address(self, amount, label):
+        n = self.get_next_address_index()
+        xpub = bip32.derive(xkey=config.zpub, der_path="/0/{}".format(n))
```

Comment:
This is non-standard derivation path, bad. Normal use case would be to specify extended public key in SatSale config and then use any compatible wallet for receiving. Standard derivation path for P2WPKH specified in [BIP84](https://github.com/bitcoin/bips/blob/master/bip-0084.mediawiki) for account 0 receiving addresses is "m/84'/0'/0'/0/...". Either that one should be used or it could be another configuration option, if people want to change that (there are potential use cases, for example, if I want to receive to non-zero mixdepth (account) of JoinMarket wallet).
```suggestion
        xpub = bip32.derive(xkey=config.zpub, der_path="m/84'/0'/0'/0/{}".format(n))
```

Reply:
Not sure what I was doing here. I'm going to add some test vectors for this. Also will add derivation path to `config.py` 👍